activestorage/app/jobs/active_storage/transform_job.rb

```diff
@@ -7,6 +7,6 @@ class ActiveStorage::TransformJob < ActiveStorage::BaseJob
   retry_on ActiveStorage::IntegrityError, attempts: 10, wait: :polynomially_longer
 
   def perform(blob, transformations)
-    blob.variant(transformations).processed
+    blob.send(blob.previewable? ? :preview : :variant, transformations).processed
```

Comment:
[`representation`](https://api.rubyonrails.org/v7.1/classes/ActiveStorage/Blob/Representable.html#method-i-representation) implements this logic internally.

```suggestion
    blob.representation(transformations).processed
```